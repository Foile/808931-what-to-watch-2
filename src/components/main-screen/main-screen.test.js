import React from 'react';
import renderer from "react-test-renderer";
import MainScreen from './main-screen';

it(`App correctly renders after relaunch`, () => {
  const movies = [
    {id: 1, name: `Macbeth`, previewImage: ``, videoLink: ``},
    {id: 2, name: `Aviator`, previewImage: ``, videoLink: ``}
  ];
  const tree = renderer
  .create(<MainScreen movies = {movies} />
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
