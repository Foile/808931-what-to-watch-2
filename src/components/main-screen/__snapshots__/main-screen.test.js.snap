// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="background Macbeth"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Macbeth poster"
            height="327"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy
            </span>
            <span
              className="movie-card__year"
            />
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                poster=""
                src=""
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Macbeth
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                poster=""
                src=""
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Aviator
            </h3>
          </a>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
